baseapp/abci.go

```diff
@@ -266,8 +299,14 @@ func (app *BaseApp) PrepareProposal(req abci.RequestPrepareProposal) abci.Respon
 // Ref: https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-060-abci-1.0.md
 // Ref: https://github.com/tendermint/tendermint/blob/main/spec/abci/abci%2B%2B_basic_concepts.md
 func (app *BaseApp) ProcessProposal(req abci.RequestProcessProposal) abci.ResponseProcessProposal {
-	// TODO: Implement.
-	return abci.ResponseProcessProposal{Status: abci.ResponseProcessProposal_ACCEPT}
+	if app.processProposal == nil {
+		panic("app.ProcessProposal is not set")
+	}
+
+	ctx := app.prepareProposalState.ctx.WithVoteInfos(app.voteInfos)
+	ctx = ctx.WithConsensusParams(app.GetConsensusParams(ctx))
```

Comment:
1. How come we don't do `app.deliverState.ctx = ...`?
2. We probably want to include other fields, like `Height`, `Time`, `Proposer`, and `Hash`

Reply:
> 1. How come we don't do app.deliverState.ctx = ...?

Why would we? Only the `sdk.Context` is being passed down into the ProcessProposal handler.

> 2. We probably want to include other fields, like Height, Time, Proposer, and Hash

BeginBlock: https://github.com/cosmos/cosmos-sdk/blob/a15d9f0d7208571994af5597eba02efb2ec0cc76/baseapp/abci.go#L199-L203 with Height and Hash.  I'm not sure where to source Proposer or Time from?

Reply:
Ah yes, they are part of the `RequestProcessProposal` but there's no harm to making them exist in the context.

In fact, I can imagine an app that passes the context down to further calls but doesn't want to necessarily want to also pass a `RequestProcessProposal`. Does that make sense?